src/app/views/users/UsersList.jsx

```diff
@@ -0,0 +1,174 @@
+                      type: "warning",
+                      message: "An error's occurred whilst trying to get users. You may only be able to see previously loaded information.",
+                      isDismissable: true,
+                  };
+                  notifications.add(notification);
+                  break;
+              }
+              case "UNEXPECTED_ERR": {
+                  const notification = {
+                      type: "warning",
+                      message:
+                        "An unexpected error's occurred whilst trying to get users. You may only be able to see previously loaded information.",
+                      isDismissable: true,
+                  };
+                  notifications.add(notification);
+                  break;
+              }
+              case "FETCH_ERR": {
+                  const notification = {
+                      type: "warning",
+                      message:
+                        "There's been a network error whilst trying to get users. You may only be able to see previously loaded information.",
+                      isDismissable: true,
+                  };
+                  notifications.add(notification);
+                  break;
+              }
+              default: {
+                  log.event("Unhandled error fetching users", log.data({ status_code: error.status }), log.error(error));
+                  const notification = {
+                      type: "warning",
+                      message:
+                        "An unexpected error's occurred whilst trying to get users. You may only be able to see previously loaded information.",
+                      isDismissable: true,
+                  };
+                  notifications.add(notification);
+                  break;
+              }
+          }
+          console.error("Error getting all users:\n", error);
+      });
+};
+
+export const mapUserToState = (rootPath, user) => {
+    try {
+        const id = user.email;
+        const title = `${user.forename} ${user.lastname}`;
+        const details = [user.email];
+        const url = `${rootPath}/users/${id}`;
+        return { ...user, id, title, details, url };
+    } catch (error) {
+        const notification = {
+            type: "warning",
+            message: "Error mapping users to state",
+            isDismissable: true,
+            autoDismiss: 3000,
+        };
+        notifications.add(notification);
+        console.error("Error mapping users to state: ", error);
+        log.event("Error mapping users to state", log.error(error));
+        return false;
+    }
+};
+
+const propTypes = {
+    rootPath: PropTypes.string.isRequired,
+    dispatch: PropTypes.func.isRequired,
+    children: PropTypes.element,
+    params: PropTypes.shape({
+        userID: PropTypes.string,
+    }).isRequired,
+    loggedInUser: PropTypes.object,
+    users: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
```

Comment:
I think we know the fields of the users object, so we should prob populate the shape.

Reply:
Fields added.